CONTRIBUTING.md

```diff
@@ -0,0 +1,203 @@
+# Contributing to vsomeip-lib
+
+All types of contributions are encouraged and valued.
+
+## Did you find a bug?
+
+- **Ensure the bug was not already reported** by searching on GitHub under [Issues](https://github.com/COVESA/vsomeip/issues).
+
+- If you cannot find an existing open issue addressing the problem, feel free to create a new one [here](https://github.com/COVESA/vsomeip/issues/new). Make sure to provide a descriptive **title, a clear explanation of the issue, relevant information**, and if possible, include a code sample or a test case demonstrating the expected behavior that is not occurring.
+
+## Styleguides
+
+We prioritize consistency and maintainability in our code bases. To achieve this, we have established comprehensive coding standards encompassing general style guidelines and specific configurations for Clang-Format integration.
+
+Explore the [Table of Contents](#table-of-contents) to discover various paths for assistance and comprehensive insights into our coding standards.
+
+### Table of Contents
+
+- [General Guidelines](#general-guidelines)
+
+- [Naming Conventions](#naming-conventions)
+
+- [Clang Format Integration](#clang-format-integration)
+
+  - [Based Style](#based-style)
+
+  - [Standard](#standard)
+
+  - [Clang-Format Customization](#clang-format-customization)
+
+  - [How to use Clang Format](#how-to-use-clang-format)
+
+  - [Workflow Summary](#workflow-summary)
+
+- [Showcase of good and bad practice](#showcase-of-good-and-bad-practice)
+
+### General Guidelines
+
+Before submitting a pull request, please ensure that you adhere to the following coding guidelines.
+
+### Naming Conventions
+
+Maintaining a consistent naming convention enhances code readability and comprehension. Our naming conventions dictate the following:
+   - **Arguments:** Argument names should start with an underscore.
+
+   - **Members:** Member variables should end with an underscore.
+
+   - **Local Variables:** Local variables neither start nor end with an underscore.
+
+   - **Static Variables:** Static variables should end with two underscores.
+
+### Clang Format Integration
+
+Incorporating Clang Format into our projects is integral to ensuring uniform and readable code formatting. Clang Format is a powerful tool that automatically formats C, C++, and Objective-C code based on predefined style guidelines.
+
+#### Based Style
+
+```yaml
+BasedOnStyle: WebKit
+```
+Our code style is based on the WebKit style, which is in turn influenced by the Qt style.
+
+Webkit style guidelines can be found here [Webkit.org](https://www.webkit.org/code-style-guidelines/)
+
+#### Standard
+
+```yaml
+Standard: c++14
+```
+The C++ standard to be used, in this case, C++14.
```

Comment:
c++17 (see https://github.com/COVESA/vsomeip/commit/cf497232adf84f55947f7a24e1b64e04b49f1f38 )

Reply:
done in https://github.com/COVESA/vsomeip/pull/733/commits/d452c10902b95ffac25c651fe0bc8f2a68ef4515